minpy/array_variants/numpy/numpy_core.py

```diff
@@ -20,6 +25,63 @@ def _minpy_getitem_grad(arr, index, g):
     return ret
 
 
+def _minpy_amax_amin_grad(ans, a, axis, out, keepdims):
+    """ Gradient of amax function """
+    repeater, _ = _match_shape(a, axis, keepdims=keepdims)
+    argmax_locations = a == repeater(ans)
+    return lambda g: repeater(g) * argmax_locations
+
+
+def _match_shape(a, axis, keepdims):
+    """ Return a function repeats input to match a given shape
```

Comment:
Is this equal to broadcasting? If so, just use broadcasting to avoid duplicate logics.